Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/AssetPlatformComponentRemover.cpp

```diff
@@ -54,8 +54,8 @@ namespace AzToolsFramework::Prefab::PrefabConversionUtils
                     {
                         // Loop over an entity's components backwards and pop-off components that shouldn't exist.
                         AZStd::vector<AZ::Component*> components = entity->GetComponents();
-                        const int oldComponentCount = components.size();
-                        for (int i = oldComponentCount - 1; i >= 0; --i)
+                        unsigned __int64 oldComponentCount = components.size();
```

Comment:
This variable should be changed to size_t.
`__int64` is a Windows only concept and this code would not compile on other platforms.
Furthermore the body of for loop needs to be adjust to subtract 1 from the `i` variable to prevent an issue with underflow when an entity has zero components.
```c++
                        size_t oldComponentCount = components.size();
                        for (size_t i = oldComponentCount ; i > 0; --i)
                        {
                            AZ::Component* component = components[i - 1];
```